fsm.go

```diff
@@ -237,3 +246,7 @@ func (r *fsmS) cpuSetFileContent(pid int) string {
 
 	return string(data)
 }
+
+func expDelay(retryNumber int) time.Duration {
+	return time.Duration(math.Pow(2, float64(retryNumber-1))) * exponentialRetryPeriodBase
```

Comment:
Just curious why do we always need to add 1 to the retry number when calling this function to subtract it later?

Reply:
Actually good question :) I wanted to have a variable with a clear name - `retryNumber`. So for a caller, it makes sense: try the first time, second, etc. (2^0, 2^1, 2^2 ...) so I do minus 1. On the caller side, I have to calculate the number of current retry. In a way how "fsm" works now, without changing a logic I have to do `retryNumber := maximumRetries - r.retriesLeft + 1`. I plan to change this logic when will work on simplification/removal of "fsm". This is a plan.